### PR TITLE
added invisible man

### PR DESCRIPTION
**Pull request by ChaoticNeutralGnome**
super minor bug with this role: for some reason I can't seem to delete the variable that stores the target no matter what I do, so if invis votes no one the next day it will still go on the old target.

### CODE REVIEW

**Comment by pikulet** [comment]
**invisible man tests**
- [ ] can sit in meeting
- [ ] can sit in multiple meetings (party, mafia?) does it sit in party
- [ ] chosen target is voted off
- [ ] voting "no one"

**Comment by pikulet** [comment]
"it will still go on the old target" this mechanic is "accepted" in other roles 
- oracle
- actress

I think we should add a function that runs when "no one" is selected. It will solve a lot of roles that rely on saving data to the role.

File: Games/types/Mafia/roles/Village/InvisibleMan.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const Role = require("../../Role");
+
+module.exports = class Villager extends Role {
```
Comment (pikulet) [inline Games/types/Mafia/roles/Village/InvisibleMan.js:3]: classname can be fixed

File: Games/types/Mafia/const/Priority.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ const PRIORITY_WIN_BY_LYNCHING = -10
 const PRIORITY_WIN_CHECK_DEFAULT = 0
 
 // DAY PRIORITY
+const PRIORITY_DAY_MEETING_DEFAULT = -10
```
Comment (pikulet) [inline Games/types/Mafia/const/Priority.js:12]: currently not exported but it does not matter, it's exported in chef PR too

File: Games/types/Mafia/roles/cards/Eavesdrop.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                    labels: ["eavesdrop"],
+                    priority: PRIORITY_DAY_MEETING_DEFAULT,
+                    run: function () {
+                        this.actor.role.data.stalk = this.target;
+                    }
+                }
+            },
+            "Eavesdropping": {
+                states: ["Night"],
+                flags: ["anonymous", "speech"],
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/Eavesdrop.js:23]: you need a "voting" flag if you want input type boolean

right now there is no option for the invisible man to vote yes/no

Comment (pikulet) [reply Games/types/Mafia/roles/cards/Eavesdrop.js:23]: a more appropriate input type might be a single-option "End Meeting" , which illuminati can also use

File: Games/types/Mafia/roles/cards/Eavesdrop.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                },
+            }
+        };
+        this.listeners = {
+            "message": function (message) {
+                if (this.game.getStateName() === "Night" &&
+                message.meeting &&
+                this.game.getMeetingByName("Eavesdropping") &&
+                message.meeting.hasJoined(this.data.stalk) &&
+                message.meeting.name !== "Eavesdropping"){
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/Eavesdrop.js:43]: maybe party chat should be intentionally excluded else the Eavesdrop meeting will just be the party chat spamming it

Comment (ChaoticNeutralGnome) [reply Games/types/Mafia/roles/cards/Eavesdrop.js:43]: I meant to make it only meetings that the Invisible man isn't in. I can update that

**Comment by ChaoticNeutralGnome** [comment]
Invisible man now only get's messages from meetings they aren't in

**Review by DrSharky — APPROVED** [review]

**Review by pikulet — APPROVED** [review]
you can use mustAct to sweep the "novote" issue under

**Comment by pikulet** [comment]
you can use mustAct to sweep the "novote" issue under

File: Games/types/Mafia/roles/cards/Eavesdrop.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+const Message = require("../../../../core/Message");
+module.exports = class Eavesdrop extends Card {
+
+    constructor(role) {
+        super(role);
+
+        this.meetings = {
+            "Eavesdrop On": {
+                states: ["Day"],
+                flags: ["voting"],
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/Eavesdrop.js:12]: you can add "mustAct" to sweep the "novote" issue under

Comment (ChaoticNeutralGnome) [reply Games/types/Mafia/roles/cards/Eavesdrop.js:12]: created an action to clear the stalk target each night to resolve this issue

**Review by pikulet — APPROVED** [review]

**Review by DrSharky — APPROVED** [review]

**Pull request merged**